Obsidian/Plugins/PluginProviders/PackedPluginProvider.cs

```diff
@@ -0,0 +1,250 @@
+﻿using Microsoft.Extensions.Logging;
+using Obsidian.API.Plugins;
+using Org.BouncyCastle.Crypto;
+using System.Collections.Frozen;
+using System.IO;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Obsidian.Plugins.PluginProviders;
+public sealed class PackedPluginProvider(PluginManager pluginManager, ILogger logger)
+{
+    private readonly PluginManager pluginManager = pluginManager;
+    private readonly ILogger logger = logger;
+
+    public async Task<PluginContainer?> GetPluginAsync(string path)
+    {
+        await using var fs = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read);
+        using var reader = new BinaryReader(fs);
+
+        var header = Encoding.ASCII.GetString(reader.ReadBytes(4));
+        if (header != "OBBY")
```

Comment:
This should also work (and be more efficient)
```cs
var header = reader.ReadBytes(4);
if ("OBBY"u8.SequenceEqual(header))
```

Reply:
yessir